### DISCUSSION TITLE
Error handler don't log ignored

### DISCUSSION THREAD

**Discussion by ossigeno** [mako-framework/framework#324] [Q&A]
In my application.php config file I have

```
        'error_handler' =>
	[
		'log_errors'     => true,
		'display_errors' => false,
		'keep'           => ['headers' => ['Access-Control-.*']],
		'dont_log'       =>
		[
			mako\http\exceptions\NotFoundException::class,
		],
        ]
```
Despite this my mako logs are getting filled with 404 errors from the various hack attempts which makes finding genuine logging difficult. Is there something I am missing to stop NotFoundExceptions being logged?

[2024-10-13T05:46:50.694617+01:00] mako.ERROR: [bd36fc1e-caac-4719-915c-9d1849e17dfc@41058] The resource you requested could not be found. It may have been moved or deleted.
Uri: /phpMyAdmin-4.8
 {"user":{"ip_address":"172.71.159.17"}}

**Comment by freost** [comment]
I set up and new app and replaced the default `mako\http\exceptions\HttpStatusException::class` with `mako\http\exceptions\NotFoundException::class` and I'm not getting any 404 entires in my logs.

Are you using a custom logger or error handler? The log format doesn't seem to match the default log format I'm getting when I test.

**Reply by ossigeno** [reply]
Ah that could be it. I do have some logic in bootstrap.php to set the formatting and also email me any non 404 errors.

$server_name = isset($_SERVER['SERVER_NAME']) ? $_SERVER['SERVER_NAME'] : ""; 
$logger = $container->get('logger');

$syslogHandler = new \Monolog\Handler\SyslogHandler($server_name, 'local0', Monolog\Level::Info);
$formatter = new \Monolog\Formatter\LineFormatter("%message%\n");
$syslogHandler->setFormatter($formatter);
$logger->getLogger()->pushHandler($syslogHandler);

\app\services\LoggerHelper::addTag($server_name);

$errorHandler->handle('\Exception', function($e) use ($server_name)
{
    \app\services\LoggerHelper::write($e->getMessage(), \app\models\types\LogLevel::Error);

    $userAgent = isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : "";

    if($e->getCode() != 404 && $userAgent != "Mediapartners-Google") {
        $body = "<table cellpadding='5' border='1' style='border-collapse:collapse;'>";
        $body .= "<tr><td>ip:</td><td>" . \app\services\IPAddressHelper::get_client_ip() . "</td></tr>";
          
        if(isset($_SERVER['HTTP_USER_AGENT'])) {
            $body .= "<tr><td>user agent:</td><td>" . $_SERVER['HTTP_USER_AGENT'] . "</td></tr>";
        }
        if(isset($_SERVER['REQUEST_METHOD'])) {
            $body .= "<tr><td>method:</td><td>" . $_SERVER['REQUEST_METHOD'] . "</td></tr>";
        }
        if(isset($_SERVER['REQUEST_URI'])) {
            $body .= "<tr><td>request:</td><td>" . $_SERVER['REQUEST_URI'] . "</td></tr>";
        }
        $body .= "<tr><td>code:</td><td>" . $e->getCode() . "</td></tr>";
        $body .= "<tr><td>line:</td><td>" . $e->getLine() . "</td></tr>";
        $body .= "<tr><td>file:</td><td>" . $e->getFile() . "</td></tr>";
        $body .= "</table><br/><hr/><br/>";
        $body .= str_ireplace("#", "<br/>#", $e->getTraceAsString());
        \app\services\NewEmailer::SendMail($email, [], "Exception on $server_name - " . $e->getMessage(), "", $body);
    }
});
